src/main/java/hudson/plugins/emailext/plugins/content/FailedTestsContent.java

```diff
@@ -145,4 +156,12 @@ private int outputTest(StringBuilder buffer, TestResult failedTest,
     private String getLineBreak() {
         return escapeHtml ? "<br/>" : "\n";
     }
+
+    private List<TestResult> filterTests(List<? extends TestResult> failedTests, String regexPattern) {
+        if(regexPattern.length() != 0) {
+            Pattern pattern = Pattern.compile(regexPattern);
+            failedTests = failedTests.stream().filter(t -> pattern.matcher(t.getFullName()).matches()).collect(Collectors.toList());
+        }
+        return (List<TestResult>)failedTests;

```

Comment:
Instead of casting, I would prefer to use the same return type.

Reply:
That would make the `failedTests` in `evaluate` also a `List<? extends TestResult>` which is ambiguous. Does it make sense to type cast it in `evaluate` ?

Reply:
You mean the type of `failedAndFilteredTests` in `evaluate`? Not sure what ambiguity are you talking about.

Reply:
The old code was for (TestResult failedTest : testResult.getFailedTests()) { and the return type of testResult.getFailedTests() was List<? extends TestResult>, so if you keep the same type everywhere, there should be no casting involved, right?

Reply:
yeah, by ambiguity i meant the `List<? extends>` part. Previously, its type was not explicitly mentioned anywhere.